main/src/main/java/ru/practicum/ewm/service/impl/PlaceLocationServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package ru.practicum.ewm.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import ru.practicum.ewm.exception.EntityNotFoundException;
+import ru.practicum.ewm.mapper.PlaceLocationMapper;
+import ru.practicum.ewm.model.PlaceLocation;
+import ru.practicum.ewm.model.dto.PlaceLocationDto;
+import ru.practicum.ewm.model.dto.PlaceLocationUpdateDto;
+import ru.practicum.ewm.repository.PlaceLocationRepository;
+import ru.practicum.ewm.service.PlaceLocationService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+@RequiredArgsConstructor
+public class PlaceLocationServiceImpl implements PlaceLocationService {
+    public final PlaceLocationMapper placeLocationMapper;
+    private final PlaceLocationRepository placeLocationRepository;
+
+    @Override
+    public PlaceLocationDto savePlaceLocation(PlaceLocationDto placeLocationDto) {
+        PlaceLocation placeLocation = placeLocationMapper.mapToEntity(placeLocationDto);
+        return placeLocationMapper.mapToDto(placeLocationRepository.save(placeLocation));
+    }
+
+    @Override
+    public PlaceLocationDto updatePlaceLocation(Long locId, PlaceLocationUpdateDto placeLocationUpdateDto) {
```

Comment:
В этом методе вы проверяете каждое поле объекта PlaceLocationUpdateDto на null, прежде чем устанавливать его в объект PlaceLocation. Это хорошо, но вы могли бы улучшить свой код, используя Java Optional API для обработки null значений. Например, вы можете использовать метод Optional.ifPresent():

```
Optional.ofNullable(placeLocationUpdateDto.getName()).ifPresent(placeLocation::setName);
Optional.ofNullable(placeLocationUpdateDto.getLon()).ifPresent(placeLocation::setLon);
Optional.ofNullable(placeLocationUpdateDto.getLat()).ifPresent(placeLocation::setLat);
Optional.ofNullable(placeLocationUpdateDto.getRadius()).ifPresent(placeLocation::setRadius);
```

Reply:
Возможно ли оборачивать подобным образом следующие конструкции?
```
if (eventDto.getCategory() != null)
            updatedEvent.setCategory(
                    categoryMapper.mapToCategory(
                            categoryService.getCategory(eventDto.getCategory())));
```

```
if (updatedEvent.getPublishedOn() != null) {
            throw new EventAlreadyPublishedException(String.format("Cannot update event with id=%d because it's not in the right state: PUBLISHED", eventId));
        }
```